src/accelerate/accelerator.py

```diff
@@ -149,10 +175,14 @@ def __init__(
         # Mixed precision attributes
         self.scaler = None
         self.native_amp = False
-        if self.state.use_fp16:
+        if self.state.mixed_precision == "fp16":
             self.native_amp = version.parse(torch.__version__) >= version.parse("1.6")
             kwargs = self.scaler_handler.to_kwargs() if self.scaler_handler is not None else {}
             self.scaler = torch.cuda.amp.GradScaler(**kwargs)
+        elif self.state.mixed_precision == "bf16":
+            self.native_amp = version.parse(torch.__version__) >= version.parse("1.10")
+            kwargs = self.scaler_handler.to_kwargs() if self.scaler_handler is not None else {}
+            self.scaler = torch.cuda.amp.GradScaler(**kwargs)
```

Comment:
I realize we are silently not training in mixed precision when:
- `mixed_precision="fp16"` and torch version is < 1.6
- `mixed_precision="bf16"` and torch version is < 1.10

I think we should raise an error message in both cases, or at least a warning.